src/preferences/dialog/dlgprefwaveform.cpp

```diff
@@ -24,11 +24,15 @@ DlgPrefWaveform::DlgPrefWaveform(QWidget* pParent, MixxxMainWindow* pMixxx,
 
     // Populate waveform options.
     WaveformWidgetFactory* factory = WaveformWidgetFactory::instance();
+    // We assume that the 'handles' origin list remains constant.
+    // We will use the handles index later on to set waveform types and to
+    // update the combobox.
     QVector<WaveformWidgetAbstractHandle> handles = factory->getAvailableTypes();
     for (int i = 0; i < handles.size(); ++i) {
-        waveformTypeComboBox->addItem(handles[i].getDisplayName(),
-                                      handles[i].getType());
+        waveformTypeComboBox->addItem(handles[i].getDisplayName(), i);
     }
+    // Sort the combobx itens alphabetically
+    waveformTypeComboBox->model()->sort(0);
```

Comment:
Nice. I forgot about this simple option after I didn't find a sort method in QComboBox :see_no_evil: